runtime/doc/develop.txt

```diff
@@ -443,6 +444,59 @@ with a {noun} that groups functions under a common concept).
   and uses the "del" {verb}.
 
 
+INTERFACE PATTERNS                                      *dev-patterns*
+
+Prefer adding a single `nvim_{topic}_{verb}_…` interface for a given topic.
+
+Example: >
+
+    nvim_ns_add(
+      ns_id: int,
+      filter: {
+        handle: integer (buf/win/tabpage id)
+        scope: "global" | "win" | "buf" | "tabpage"
+      }
+    ): { ok: boolean }
+
+    nvim_ns_get(
+      ns_id: int,
+      filter: {
+        handle: integer (buf/win/tabpage id)
+        scope: "global" | "win" | "buf" | "tabpage"
```

Comment:
Where does this filter->handle/scope convention come from and is it really better than opts->buf/win/tabpage which is used all over the place already? I think omission of all three is unambiguous enough to mean global scope? (No strong opinion but mainly asking to know whether I should update nvim_redraw() with this pattern.)

Reply:
> Where does this `filter->handle/scope` convention

From `nvim_set_option_value()` I guess, like you mentioned. It might make sense for options but is scope` really necessary for most APIs that just need to target `win/buf/tabpage`?

Reply:
> It might make sense for options but is scope really necessary for most APIs that just need to target `win/buf/tabpage`

How else would they do it? Adding triple or quadruple functions (nvim_win/nvim_buf/nvim_tabpage) isn't really gaining anything, while adding a lot of noise.

Reply:
> How else would they do it?

`opts->win/buf/tabpage` as opposed to `opts->filter->global/win/buf/tabpage` which seems to be the proposal here. Wasn't suggesting separate functions but merely keeping with what I thought was an already established convention.

Reply:
Oh, yeah. Either way is fine. `opts->win/buf/tabpage` can be a bit awkward because of the "mutual exclusion". However, it's arguably more readable, less fussy, and pretty common already. So I'll update the doc.